src/main/java/org/tuberlin/de/geodata/MapCoordToDistrict.java

```diff
@@ -40,176 +28,151 @@
-				feats.add(d);
-			}
-			return feats;
-
-		} catch (MalformedURLException e) {
-			e.printStackTrace();
-		} catch (IOException e) {
-			e.printStackTrace();
-		} finally {
-			features.close();
-			dataStore.dispose();
-		}
-
-
-		return null;
-	}
-
-
-
-  	public static class DistrictMapper extends RichMapFunction<Taxidrive, Taxidrive> {
-
-		private Collection<District> districtGeometries;
-
-		@Override
-		public void open(Configuration parameters) {
-			this.districtGeometries = getRuntimeContext().getBroadcastVariable("districtGeometries");
-		}
-
-
-		@Override
-		public Taxidrive map(Taxidrive taxidrive) throws Exception {
-
-			for(District feat : districtGeometries) {
-
-				// pickup location
-				if (coordinateInRegion(feat.geometry, new Coordinate(taxidrive.dropoff_longitude, taxidrive.dropoff_latitude))) {
-					taxidrive.setPickupDistrict(feat.district);
-					break;
-				}
-			}
-
-			for(District feat : districtGeometries) {
-				// dropoff location
-				if(coordinateInRegion(feat.geometry, new Coordinate(taxidrive.dropoff_longitude, taxidrive.dropoff_latitude))){
-					taxidrive.setDropoffDistrict(feat.district);
-					break;
-				}
-			}
-			return taxidrive;
-		}
-	}
-
-
-	public static boolean coordinateInRegion(Coordinate[] region, Coordinate coord) {
-		// adapt the code from http://stackoverflow.com/questions/12083093/how-to-define-if-a-determinate-point-is-inside-a-region-lat-long
-		int i, j;
-		boolean isInside = false;
-		//create an array of coordinates from the region boundary list
-		int sides = region.length;
-		for (i = 0, j = sides - 1; i < sides; j = i++) {
-			//verifying if your coordinate is inside your region
-			if (
-					(
-							(
-									(region[i].x <= coord.x) && (coord.x < region[j].x)
-							) || (
-									(region[j].x) <= coord.x) && (coord.x < region[i].x)
-							)
-					 &&
-							(coord.y < (region[j].y - region[i].y) * (coord.y - region[i].x) / (region[j].x - region[i].x) + region[i].y)
-					) {
-				isInside = !isInside;
-			}
-		}
-		return isInside;
-	}
+    public static void main(String[] args) throws Exception {
+        // set up the execution environment
+        final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+        //ShapeToTextConverter.convertShape("data/manhattan_districts.shp", "data/districts");
+
+        // load taxi data from csv-file and map districts
+        DataSet<Taxidrive> taxidrives = readData(env, "data/testData.csv");
+
+        //taxidrives.writeAsText("data/testDataWithDistricts");
+        taxidrives.print();
+
+
+        // execute program
+        //env.execute("Flink Java API Skeleton");
+    }
+
+    public static DataSet<Taxidrive> readData(ExecutionEnvironment env, String dataPath) throws Exception {
+        // set up the execution environment
+        // load taxi data from csv-file
+        DataSet<Taxidrive> taxidrives = env.readCsvFile(dataPath)
```

Comment:
Just one question: the reading of the data seems to be in this class as well as in BestTaxidriver.class. Seems to be exactly the same code. Why don't you reuse the code?